spec/aliases/unixpath_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+if Puppet.version.to_f >= 4.5
+  describe 'test::unixpath', type: :class do
+    describe 'valid paths handling' do
+      %w{
+        /usr2/username/bin:/usr/local/bin:/usr/bin:.
+        /var/tmp
+        /Users/helencampbell/workspace/puppetlabs-stdlib
+      }.each do |value|
+        describe value.inspect do
+          let(:params) {{ value: value }}
+          it { is_expected.to compile }
+        end
+      end
+    end
+
+    describe 'invalid path handling' do
+      context 'garbage inputs' do
+        [
+          nil,
+          [ nil ],
+          [ nil, nil ],
+          { 'foo' => 'bar' },
+          { },
+          '',
+          "C:/whatever"
```

Comment:
These tests should contain at least one testcase with wrong slashing, and one with a NUL character.